src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -920,64 +924,69 @@ export class BundleDataClient {
             fillCounter++;
             const relayDataHash = getRelayEventKey(fill);
             if (v3RelayHashes[relayDataHash]) {
-              if (!v3RelayHashes[relayDataHash].fill) {
-                assert(
-                  isDefined(v3RelayHashes[relayDataHash].deposits) && v3RelayHashes[relayDataHash].deposits!.length > 0,
-                  "Deposit should exist in relay hash dictionary."
+              if (v3RelayHashes[relayDataHash].fill) {
```

Comment:
The diff here is large because we're now reversing the if-else logic. First we check if the fill is a duplicate, and then throw, otherwise continue with the rest of the logic that was formerly in the `if` statement.